lib/parser/index.js

```diff
@@ -82,7 +83,14 @@ function parse(juttle, options) {
 
         return Promise.resolve(module.source)
             .then(function(juttle) {
-                return doParse(juttle, _.extend(options, { filename: module.name }));
+                // If the module has a filename property, it was
+                // passed back from the resolver as the full path to
+                // the file. Use it instead.
+                let filename = module.name;
+                if (_.has(module, 'filename')) {
+                    filename = module.filename;
+                }
```

Comment:
I think it would be cleaner if the protocol was that a resolver always sets `module.filename`. Then we could rely on that here.